### ISSUE TITLE
Reworking diffing in Test2

### ISSUE THREAD

**Issue by martindholmes** [TEIC/Stylesheets#427]
@martindholmes and @sydb 2020-04-10, working in response to 
pull request 426:

A rethink of the way things [should] work suggests that we should:

1. Change the cleanForDiff target in build_utilities.xml so that it:

    a) incorporates @sydb's better whitespace normalization from the 
        pull request
        
    b) saves the results of the diff not over the original file, but to a 
        tempForDiff/outputFiles directory.
        
    c) performs the same operation on the corresponding expected-results
        file, and puts that result in tempForDiff/expected-results
        
    d) diffs those two files. We should massage the output format of the 
        diff tool so that it marks "output" and "expect" instead of + and -, 
        and suppresses line numbers.
        
    e) suggests that the person responding to the problem do an XML diff
        on the actual expected-results and output file, possibly turning on 
        "Ignore whitespace" or the equivalent.
        
2. Change the diffFile target so that it diffs the versions of the two files in 
    temp/outputFiles and temp/expected-results. If those files match, it 
    should go ahead and delete them.
    
3. Add a final cleanup routine which deletes the tempForDiff folder if everything
    passes.
    
This may add to build time; the last Jenkins build took 9.29, so compare when this is all done to see what the time penalty is.

**Comment by martindholmes** [comment]
Diff configuration:

`diff --old-line-format='output: %L' --new-line-format='expect: %L' --unchanged-line-format='' outputFile expectedFile`

**Comment by martindholmes** [comment]
@sydb I've tested this in a branch, and this is what I've concluded:

1. Removing all runs of whitespace and replacing them with a single u+0020 means that the resulting diff is useless because of course the files are all on one line.
1. My original approach with diff -u (ignore whitespace length differences) combined with indenting the output of cleanForDiff is actually pretty solid when it comes to detecting meaningful differences.
1. The change to the identity transform template in your pull request seems to be aimed at removing comments, but I was already matching and suppressing comments in the XSLT above it; perhaps it has another purpose?

I think the valuable thing we should use is the diff configuration values above, which does really help assuming there are lots of useful linebreaks. I would do it the other way round, though -- expected first, found next.
